extensions/podman/package.json

```diff
@@ -95,6 +107,88 @@
           "description": "Machine with root privileges"
         }
       }
+    },
+    "onboarding": {
+      "title": "Podman Setup",
+      "steps": [
+        {
+          "id": "checkPodmanInstalled",
+          "label": "Check Podman",
+          "title": "Checking for Podman installation",
+          "command": "podman.onboarding.checkPodmanInstalled",
+          "completionEvents": [
+            "onCommand:podman.onboarding.checkPodmanInstalled"
+          ]
+        },
+        {
+          "id": "startPodmanInstallation",
+          "label": "Start Setup",
+          "title": "We could not find Podman. Let's install it!",
+          "when": "onContext:podmanIsNotInstalled == true"
+        },
+        {
+          "id": "checkPodmanRequirements",
+          "label": "Check System",
+          "title": "Checking for system requirements to install Podman",
+          "when": "onContext:podmanIsNotInstalled == true",
+          "command": "podman.onboarding.checkPodmanRequirements",
+          "completionEvents": [
+            "onCommand:podman.onboarding.checkPodmanRequirements"
+          ]
+        },
+        {
+          "id": "missingRequirementView",
+          "label": "Fix System",
+          "title": "Some system requirements are missing",
+          "when": "onContext:requirementsStatus == failed && onContext:podmanIsNotInstalled == true",
+          "completionEvents": [
+            "onContext:requirementsStatus == ok"
+          ],
+          "content": [
+            [
+              {
+                "value": "${onContext:warningsMarkdown}",
+                "highlight": true
+              }
+            ],
+            [
+              {
+                "value": "When possible, we've provided information on how to address these requirements."
+              }
+            ],
+            [
+              {
+                "label": "Check requirements again",
+                "command": "podman.onboarding.checkPodmanRequirements",
+                "component": "button"
+              }
+            ]
+          ]
+        },
+        {
+          "id": "installPodmanView",
+          "label": "Install Podman",
+          "title": "Installing Podman",
+          "description": "Once installed, we will enable and configure the extension",
+          "when": "onContext:podmanIsNotInstalled == true",
+          "command": "podman.onboarding.installPodman",
+          "completionEvents": [
+            "onCommand:podman.onboarding.installPodman"
+          ]
+        },
+        {
+          "id": "podmanFailedInstallation",
+          "label": "Done",
+          "title": "Failed installing Podman",
+          "when": "onContext:podmanIsNotInstalled == true"
```

Comment:
the `onContext` prefix is needed to identify a variable that needs to be looked for in the context (like the suggestion you gave me to add a helper function `onboarding()`). I just keep using onContext to avoid increasing the complexity as we already make use of it in the completion events.

When adding a value with the 'onboarding' scope, PD adds automatically the prefix `<extension-id>.onboarding` to distiguish it from other generic context values. So we also need to take care of it when evaluating a when clause. The problem is that it is not trivial to being sure that a variable is a value or a context key so i ask the user to mark it .

Let's say there is `somevalue in somearray` or `somevalue == xyz`. How can i be sure that the user wants me to search for `somevalue`, `somearray` and `xyz` in the context? Or just search and use the context value for `somevalue` and `somearray`?

So i ask the user to mark it. 
I could guess by looking at the context. If the `<extension-id>.onboarding.xyz` does not exist in the context i use it as a value, but it looks to me more secure if the user clearly define which is a context key.

Reply:
we can also use `onboardingContext:`, i used `onContext:` so there is just one keyword to remember for the user. But if you see that it's clearer to use the former, np for me

Reply:
so `onboardingContext` or `onboarding:` (prefix with the scope of the context)

set with `setContext('key', '', scope)`, retrieve with `scope?:<key>`

Reply:
done 👍